test/functionality/dataflow/environments/resolve-tests.ts

```diff
@@ -42,4 +44,28 @@ describe('Resolve', () => {
 			});
 		});
 	});
+	describe('Builtin Constants', () => {
+		it('Unknown Identifier', () => {
+			const env = defaultEnv();
+			const result = resolvesToBuiltInConstant(undefined, env, undefined);
+			expect(result, 'should be Ternary.Never').to.be.equal(Ternary.Never);
+		});
+		it('Undefined Identifier', () => {
+			const env = defaultEnv();
+			const result = resolvesToBuiltInConstant('foo', env, undefined);
+			expect(result, 'should be Ternary.Never').to.be.equal(Ternary.Never);
+		});
+
+		const testSingle = (identifier: Identifier, expected: unknown) => function() {
+			const env = defaultEnv();
+			const result = resolvesToBuiltInConstant(identifier, env, expected);
+			expect(result, 'should be Ternary.Always').to.be.equal(Ternary.Always);
+		};
+		it('Resolve TRUE',	testSingle('TRUE', true));
+		it('Resolve T',		testSingle('T', true));
+		it('Resolve FALSE',	testSingle('FALSE', false));
+		it('Resolve F',		testSingle('F', false));
+		it('Resolve NULL',	testSingle('NULL', null));
+		it('Resolve NA',	testSingle('NA', null));
```

Comment:
Currently missing are negative tests although they may not be as interesting. Yet something like "this does not resolve to anything", "this does not resolve to what i want", or this "may resolve to what we want" - may be of interest.

Reply:
I now have tests for an Identifier which is not defined and an Identifier that does not resolve to what we wanted.
But I am not quite sure how to implement the test case for "maybe resolves to what we want".